src/main/groovy/com/vherasymenko/avro/decoder/core/CourseInstallDecoderService.groovy

```diff
@@ -1,23 +1,31 @@
 package com.vherasymenko.avro.decoder.core
 
 import com.vherasymenko.avro.decoder.core.json.AvroJsonDecoderPort
-import event.course_install.CourseInstall
+import groovy.util.logging.Slf4j
+import org.apache.avro.Schema
+import org.springframework.cloud.stream.schema.client.SchemaRegistryClient
 
 /**
  * Decoder for the course install document.
  */
+@Slf4j
 class CourseInstallDecoderService implements CourseInstallDecoderPort {
 
-
     private final AvroJsonDecoderPort decoder
 
-    CourseInstallDecoderService( AvroJsonDecoderPort aDecoder ) {
+    private final SchemaRegistryClient registryClient
+
+    CourseInstallDecoderService( AvroJsonDecoderPort aDecoder, SchemaRegistryClient aRegistryClient ) {
         decoder = aDecoder
+        registryClient = aRegistryClient
     }
 
     @Override
-    void decodeCourseInstallEvent( String event ) {
-        def schema = CourseInstall.classSchema
-        decoder.decodeEvent( schema, event )
+    void decodeCourseInstallEvent( String event, int schemaId ) {
+        def writeSchema = new Schema.Parser().parse( registryClient.fetch( schemaId ) )
+        def readSchema = new Schema.Parser().parse( new File( 'src/main/resources/avro/course_install_2.avsc' ) )
```

Comment:
Here I want to decode the event with the schema version 2. For this I need both the write and reader schema. The writer schema was saved in the schema registry by encoder. The reader schema isn't in the schema registry yet because we don't use preloading of schemas for now.